pkg/build/android.go

```diff
@@ -65,33 +62,151 @@ func (a android) readCompiler(archivePath string) (string, error) {
+			return err
+		}
+	}
+
+	// Make kernel image and prepare common kernel for modules.
+	cmd, err = a.makeCmd(params, "bzImage", "modules", "prepare-objtool")
+	if err != nil {
+		return fmt.Errorf("failed to create command to make bzImage: %v", err)
+	}
+	cmd.Dir = commonKernelDir
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return fmt.Errorf("failed to make bzImage: %v", err)
+	}
+
+	// Install common modules.
+	moduleStagingDir := filepath.Join(commonKernelDir, "staging")
+	moduleInstallFlag := fmt.Sprintf("INSTALL_MOD_PATH=%v", moduleStagingDir)
+	cmd, err = a.makeCmd(params, "bzImage", moduleInstallFlag, "modules_install")
+	if err != nil {
+		return fmt.Errorf("failed to create command to install modules: %v", err)
+	}
+	cmd.Dir = commonKernelDir
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return fmt.Errorf("failed to install modules: %v", err)
+	}
+	return nil
+}
+
+func (a android) buildExtModules(params Params) error {
+	commonKernelDir := filepath.Join(params.KernelDir, "common")
+
+	// Location of external modules relative to common kernel dir.
+	mFlag := fmt.Sprintf("M=../%v", modulePath)
+	// Absolute location of the kernel source directory.
+	srcFlag := fmt.Sprintf("KERNEL_SRC=%v", commonKernelDir)
+
+	moduleStagingDir := filepath.Join(commonKernelDir, "staging")
+	moduleInstallFlag := fmt.Sprintf("INSTALL_MOD_PATH=%v", moduleStagingDir)
+
+	// Make external modules.
+	cmd, err := a.makeCmd(params, "-C", modulePath, mFlag, srcFlag, moduleInstallFlag)
+	if err != nil {
+		return fmt.Errorf("failed to create command to make external modules modules: %v", err)
+	}
+	cmd.Dir = params.KernelDir
+	cmd.Env = append([]string{}, moduleEnvVar)
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return fmt.Errorf("failed to make external modules: %v", err)
+	}
+
+	// Install modules.
+	cmd, err = a.makeCmd(params, "-C", modulePath, mFlag, srcFlag, moduleInstallFlag, "modules_install")
+	if err != nil {
+		return fmt.Errorf("failed to create command to install modules: %v", err)
+	}
+	cmd.Dir = params.KernelDir
+	cmd.Env = append([]string{}, moduleEnvVar)
+	if err := a.runCmd(cmd, params.KernelDir); err != nil {
+		return fmt.Errorf("failed to install modules: %v", err)
+	}
+
+	return nil
+}
+
 func (a android) build(params Params) (ImageDetails, error) {
 	var details ImageDetails
+	var err error
 
 	if params.CmdlineFile != "" {
 		return details, fmt.Errorf("cmdline file is not supported for android cuttlefish images")
 	}
 	if params.SysctlFile != "" {
 		return details, fmt.Errorf("sysctl file is not supported for android cuttlefish images")
 	}
+	commonKernelDir := filepath.Join(params.KernelDir, "common")
 
-	if err := a.runBuild(params.KernelDir, kernelConfig); err != nil {
-		return details, fmt.Errorf("failed to build kernel: %s", err)
+	// Build common kernel.
+	if err := a.buildCommonKernel(params.Config, params); err != nil {
```

Comment:
Why not just pass `params` and le the `buildCommonKernel()` function read the `params.Config` field?

Reply:
Good point; changed.